data/extension/init/templates/tinygo/envoy.filters.network/default/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+var (
+	connectionCounterName = "proxy_wasm_go.connection_counter"
+	counter               proxywasm.MetricCounter
+)
+
+type context struct{ proxywasm.DefaultContext }
+
+func main() {
+	proxywasm.SetNewRootContext(func(contextID uint32) proxywasm.RootContext { return context{} })
+	proxywasm.SetNewStreamContext(func(contextID uint32) proxywasm.StreamContext { return context{} })
+}
+
+func (ctx context) OnVMStart(int) bool {
+	var err error
+	counter, err = proxywasm.DefineCounterMetric(connectionCounterName)
+	if err != nil {
+		proxywasm.LogCritical("failed to initialize connection counter: ", err.Error())
+	}
+	return true
+}
+
+func (ctx context) OnNewConnection() types.Action {
+	proxywasm.LogInfo("new connection!")
+	return types.ActionContinue
+}
+
+func (ctx context) OnDone() bool {
+	err := counter.Increment(1)
```

Comment:
hm, returning an error from `counter.Increment()` creates unnecessary overhead on the caller code.

Maybe it worth changing this API for better DevX ? (same applies to Rust)

Reply:
I know what you mean, but I am not sure hiding errors inside of the function is something Go developers like to do.

What would the expected behavior in `Increment` function be like? Something like it logs the error if the returned status is not OK? I think we should allow developers to decide what to do when error happens.

Reply:
hmm, I've changed my mind. cpp-sdk is just ignoring the returned status (https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/blob/master/proxy_wasm_api.h#L1114) so I should follow your suggestion. Thanks!